bootstrap/bootstrap-kubestellar.sh

```diff
@@ -194,67 +248,131 @@ if [ "$folder" == "" ]; then
 fi
 
 # Ensure kcp is installed
-if [ "$(kcp_installed)" == "false" ]; then
-    if [ "$verbose" != "" ]; then
-        echo "Installing kcp..."
+echo "************************************************"
+echo "Ensure kcp is installed"
+echo "************************************************"
+if [ "$(kcp_installed)" == "true" ]; then
+    echo "kcp found in the PATH at '$(which kcp)' ... skip installation."
+else
+    echo "kcp not found in the PATH."
+    if [ "$(kcp_running)" == "true" ]; then
+        echo "kcp process is running already: pid=$(pgrep kcp) ... add kcp folder to the PATH or stop kcp ... exiting."
+        exit 2
     fi
-    bash <(curl -s https://raw.githubusercontent.com/kcp-dev/edge-mc/main/bootstrap/install-kcp-with-plugins.sh) --version $kcp_version --os $os_type --arch $arch_type --ensure-folder  $folder/kcp $verbose $flagx
-    if [[ ! ":$PATH:" == *":$(get_full_path $folder/kcp/bin):"* ]]; then
-        export PATH=$PATH:$(get_full_path $folder/kcp/bin)
+    ensure_folder "$folder/kcp"
+    kcp_folder=$(get_full_path "$folder/kcp")
+    kcp_bin_folder="$kcp_folder/bin"
+    echo "Downloading kcp+plugins $kcp_version $os_type/$arch_type..."
+    kcp_download
+    echo "Installing kcp+plugins into '$kcp_folder'..."
+    kcp_install
+    if [[ ! ":$PATH:" == *":$kcp_bin_folder:"* ]]; then
+        export PATH=$kcp_bin_folder:$PATH
+        echo "Add kcp folder to the PATH: export PATH=\"$kcp_bin_folder:\$PATH\""
+        user_exports="$user_exports"$'\n'"export PATH=\"$kcp_bin_folder:\$PATH\""
     fi
 fi
 
 # Ensure kcp is running
-if [ "$(kcp_running)" == "false" ]; then
+echo "************************************************"
+echo "Ensure kcp is running"
+echo "************************************************"
+if [ "$(kcp_running)" == "true" ]; then
+    echo "kcp process is running already: pid=$(pgrep kcp) ... skip running."
+    if [ "$(kubeconfig_valid)" == "false" ]; then
+        echo "KUBECONFIG environment variable is not set correctly: KUBECONFIG='$KUBECONFIG' ... exiting!"
+        exit 3
+    fi
+    echo "Using 'KUBECONFIG=$KUBECONFIG'"
+    echo "Waiting for kcp to be ready... it may take a while"
+    until $(kcp_ready)
+    do
+        sleep 1
+    done
+    if [ "$(kcp_version)" != "$kcp_version" ]; then
+        echo "kcp running version $(kcp_version) does not match the desired version $kcp_version ... exiting!"
+        exit 4
+    else
+        echo "kcp version $(kcp_version) ... ok"
+    fi
+else
     if [ "$kcp_address" == "" ]; then
-        if [ "$verbose" != "" ]; then
-            echo "Starting kcp..."
-        fi
+        echo "Running kcp... logfile=$PWD/kcp_log.txt"
```

Comment:
Shouldn't the kcp log go under the `$kcp_folder`?  (Yes, that would have to be defined even in the non-download case.)

Reply:
Erm, ugh.  In the already-on-PATH case, the next question is whether to put the log alongside whatever `bin` holds `kcp`.  So, yeah, it's a bit irregular but perhaps the most pragmatic to just put the kcp log right in the CWD.

Reply:
I also was about to change that I was not sue if it was ok to do so.
What if kcp is pre-installed and I cannot write to that location? (/usr/local/bin)
So extatically looks better but it may have more risks or require more complex checks and foldbacks.